tests/pages/api/periodValidity.test.ts

```diff
@@ -47,11 +51,79 @@ describe('periodValidity', () => {
         });
         periodValidity(req, res);
 
-        expect(updateSessionAttributeSpy).toHaveBeenCalledWith(
-            req,
-            PERIOD_EXPIRY_ATTRIBUTE,
-            mockPeriodExpiryAttributeError,
-        );
+        expect(updateSessionAttributeSpy).toHaveBeenCalledWith(req, PERIOD_EXPIRY_ATTRIBUTE, { errors, products: [] });
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: '/periodValidity',
+        });
+    });
+
+    it('should redirect if the end of service day option selected and no time has been entered', () => {
+        const errors: ErrorInfo[] = [
+            {
+                id: 'service-end-time',
+                errorMessage: 'Specify an end time for service day',
+            },
+        ];
+
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: { fareZoneName: null },
```

Comment:
We shouldn't need to add cookieValues to mocks any more (probably worth removing them from old tests when we can as well)